scripts/zones/Southern_San_dOria/npcs/Glenne.lua

```diff
@@ -38,69 +39,37 @@ function onPath(npc)
 
 function onTrigger(player, npc)
-
-    local aSentrysPeril = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.A_SENTRY_S_PERIL)
+    local sentrysPerilStatus = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.A_SENTRY_S_PERIL)
 
     npc:wait()
-
-    if (aSentrysPeril == QUEST_AVAILABLE) then
-        player:startEvent(510)
-    elseif (aSentrysPeril == QUEST_ACCEPTED) then
-        if (player:hasItem(600) == true or player:hasItem(601) == true) then
-            player:startEvent(520)
-        else
-            player:startEvent(644)
-        end
-    elseif (aSentrysPeril == QUEST_COMPLETED) then
-        player:startEvent(521)
+    if sentrysPerilStatus == QUEST_AVAILABLE then
+        player:startEvent(510) -- Starts "A Sentry's Peril"
+    elseif sentrysPerilStatus == QUEST_ACCEPTED and (player:hasItem(600) or player:hasItem(601)) then
+        player:startEvent(520) -- Reminder to deliver the ointment to her husband.
+    elseif sentrysPerilStatus == QUEST_ACCEPTED
+        player:startEvent(644) -- "So you lost the ointment! I suppose I can give you another."
     else
-        npc:wait(0)
+        player:startEvent(521) -- "Thank you again for your help!"
     end
-
 end
 
 function onEventUpdate(player, csid, option)
 end
 
 function onEventFinish(player, csid, option, npc)
-
     npc:wait(5000)
-
-    if (csid == 510 and option == 0) then
-        if (player:getFreeSlotsCount() > 0) then
-            player:addQuest(SANDORIA, tpz.quest.id.sandoria.A_SENTRY_S_PERIL)
-            player:addItem(600)
-            player:messageSpecial(ID.text.ITEM_OBTAINED, 600)
-        else
-            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, 600) -- Dose of ointment
-        end
-    elseif (csid == 644) then
-        if (player:getFreeSlotsCount() > 0) then
-            player:addItem(600)
-            player:messageSpecial(ID.text.ITEM_OBTAINED, 600)
-        else
-            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, 600) -- Dose of ointment
-        end
-    elseif (csid == 513) then
-        if (player:getFreeSlotsCount() == 0) then
-            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, 12832) -- Bronze Subligar
-        else
-            player:tradeComplete()
-            player:addTitle(tpz.title.RONFAURIAN_RESCUER)
-            player:addItem(12832)
-            player:messageSpecial(ID.text.ITEM_OBTAINED, 12832) -- Bronze Subligar
-            player:addFame(SANDORIA, 30)
-            player:completeQuest(SANDORIA, tpz.quest.id.sandoria.A_SENTRY_S_PERIL)
-        end
+    if csid == 510 and option == 0 and npcUtil.giveItem(player, 600) then
+        player:addQuest(SANDORIA, tpz.quest.id.sandoria.A_SENTRY_S_PERIL)
+    elseif csid == 644 then
+        npcUtil.giveItem(player, 600)
+    elseif csid == 513 and npcUtil.completeQuest(player, SANDORIA, tpz.quest.id.sandoria.A_SENTRY_S_PERIL, {item = 12832, fame = 30, title = tpz.title.RONFAURIAN_RESCUER}) then
+        player:tradeComplete()
```

Comment:
If you're checking items in the trade via `npcUtil.tradeHas()`, then you should close the trade via `player:confirmTrade()` rather than `player:tradeComplete()`.

confirmTrade will consume only the items you checked for, and will return to player any excess items placed into the trade window.

tradeComplete will consume all items in the trade window.